src/main.rs

```diff
@@ -60,119 +60,117 @@ impl BFInterp {
-                            &sub.operand1,
-                            &sub.dest,
-                            BinOps::Sub,
-                        ),
-                        Instruction::Mul(mul) => self.int_bin_operations(
-                            &mul.operand0,
-                            &mul.operand1,
-                            &mul.dest,
-                            BinOps::Mul,
-                        ),
-                        Instruction::UDiv(udiv) => self.int_bin_operations(
-                            &udiv.operand0,
-                            &udiv.operand1,
-                            &udiv.dest,
-                            BinOps::Div,
-                        ),
-                        Instruction::SDiv(sdiv) => self.int_bin_operations(
-                            &sdiv.operand0,
-                            &sdiv.operand1,
-                            &sdiv.dest,
-                            BinOps::Div,
-                        ),
-                        Instruction::URem(urem) => self.int_bin_operations(
-                            &urem.operand0,
-                            &urem.operand1,
-                            &urem.dest,
-                            BinOps::Rem,
-                        ),
-                        Instruction::SRem(srem) => self.int_bin_operations(
-                            &srem.operand0,
-                            &srem.operand1,
-                            &srem.dest,
-                            BinOps::Rem,
-                        ),
-                        Instruction::FSub(fsub) => self.fl_bin_operations(
-                            &fsub.operand0,
-                            &fsub.operand1,
-                            &fsub.dest,
-                            BinOps::Sub,
-                        ),
-                        Instruction::SExt(sext) => {
-                            self.szext(&sext.operand, &sext.to_type, &sext.dest)
-                        }
-                        Instruction::FMul(fmul) => self.fl_bin_operations(
-                            &fmul.operand0,
-                            &fmul.operand1,
-                            &fmul.dest,
-                            BinOps::Mul,
-                        ),
-                        Instruction::FDiv(fdiv) => self.fl_bin_operations(
-                            &fdiv.operand0,
-                            &fdiv.operand1,
-                            &fdiv.dest,
-                            BinOps::Div,
-                        ),
-                        Instruction::ICmp(icmp) => {
-                            self.icmp(icmp.predicate, &icmp.operand0, &icmp.operand1, &icmp.dest)
-                        }
-                        Instruction::ZExt(zext) => {
-                            self.szext(&zext.operand, &zext.to_type, &zext.dest)
-                        }
-                        _ => todo!(),
+        let mut bb_name = Some(bb_name);
+        let mut bb;
```

Comment:
This doesn't need to be outside of the loop.

Reply:
Which line?

Reply:
`let mut bb`.

Reply:
Is there a difference in performance between using let inside the loop so creating a new variable on every loop vs using a mutable variable so that the same variable is changed inside the loop every time?